### PR TITLE
Add support for v8x16.swizzle and the load_splats.

### PR DESCRIPTION
**Pull request by nlewycky**
v8x16.shuffle2_imm is a rename of the previous v8x16.shuffle, but I add a copy of the code as if it were a new instruction in case the spec proposal makes further changes. The tests for old v8x16.shuffle remain in place and while there are new tests for the new v8x16.shuffle1, there are not for v8x16.shuffle2_imm. The behaviour and implementation are the same as for v8x16.shuffle, so we should simply search and replace the existing tests at some point, leaving one of them untested, probably the deprecated v8x16.shuffle.

I did test v8x16.shuffle1 against the SIMD spec test from WAVM and it passes. The WAVM spec tests for v8x16.shuffle2_imm parse but it has no invocations of the instruction.

### CODE REVIEW

**Comment by tlively** [comment]
The SIMD group decided a little while ago to rename v8x16.shuffle1 and v8x16.shuffle2_imm to v8x16.swizzle and v8x16.shuffle_imm, respectively. We just haven’t reflected that change in our proposal repo yet 😳 I’ll make that update to the proposal repo today. Sorry for the confusion.

**Comment by nlewycky** [comment]
@tlively New name, same SIMD flavour!

**Comment by tlively** [comment]
Here's the PR to update the proposal, fyi: https://github.com/WebAssembly/simd/pull/84

File: src/interp/interp-trace.cc [hunk 1]
```diff
@@ -586,6 +586,26 @@ void Thread::Trace(Stream* stream) {
           ReadU32At(pc + 4), ReadU32At(pc + 8), ReadU32At(pc + 12));
       break;
 
+    case Opcode::V8X16Swizzle:
```
Comment (binji) [inline src/interp/interp-trace.cc]: This looks to be the same as other v128 binary instructions, combine with the list above (`F64X2Mul`, etc)

**Comment by tlively** [comment]
I had misremebered: the (officially merged) new names are `v8x16.swizzle` and `v8x16.shuffle`. Apologies for the turn!

File: src/interp/interp-trace.cc (hunk 1 above)
Comment (nlewycky) [reply src/interp/interp-trace.cc]: Done.

File: test/interp/simd-load-store.txt [hunk 2]
```diff
@@ -12,6 +12,11 @@
     i32.const 4
     v128.load)
 
+  ;; i8x16 load_splat
```
Comment (binji) [inline test/interp/simd-load-store.txt]: Test other load_splat instructions too?

Comment (nlewycky) [reply test/interp/simd-load-store.txt]: Done.

**Pull request merged**